nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/is_battery_low_condition.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 Sarthak Mittal
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_BATTERY_LOW_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_BATTERY_LOW_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/float64.hpp"
+#include "behaviortree_cpp_v3/condition_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class IsBatteryLowCondition : public BT::ConditionNode
+{
+public:
+  IsBatteryLowCondition(
+    const std::string & condition_name,
+    const BT::NodeConfiguration & conf);
+
+  IsBatteryLowCondition() = delete;
+
+  BT::NodeStatus tick() override;
+
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<double>("min_battery", "Minimum battery percentage"),
+      BT::InputPort<std::string>(
+        "battery_topic", std::string("/battery_status"), "Battery topic")
+    };
+  }
+
+private:
+  void batteryCallback(std_msgs::msg::Float64::SharedPtr msg);
+
+  rclcpp::Node::SharedPtr node_;
+  rclcpp::Subscription<std_msgs::msg::Float64>::SharedPtr battery_sub_;
```

Comment:
why Float64? This is a deprecated topic in Foxy. It would be better to use standard message like [sensor_msgs::BatteryState](https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/msg/BatteryState.msg) which exposes a percentage.

Reply:
Ah great, I wasn't aware sensor_msgs::BatteryState exists.